### ISSUE TITLE
[Feature Request] Allow passing locals

### ISSUE THREAD

**Issue by cossssmin** [posthtml/posthtml-modules#34]
Just as with `posthtml-include`, it would be very useful to be able to pass in locals for the module to use.

For example:

```html
<!-- index.html -->
<module href="src/components/footer.html" locals="{'year': 2020}">
  All rights reserved.
</module>
```
```html
<!-- footer.html -->
<div>
  Copyright {{ year }} Company Name
  <content></content>
</div>
```

**Comment by anikethsaha** [comment]
it seems you can use `posthtml-expression` which will be the next plugin to this plugin and it would take care for that !

**Comment by cossssmin** [comment]
Can you give me an example of how you would pass in local variables to that module?

**Comment by anikethsaha** [comment]
With this plugin, you will resolve this module. so for input like this 


```
<!-- index.html -->
<module href="src/components/footer.html" >
  All rights reserved.
</module>
```

```
<!-- footer.html -->
<div>
  Copyright {{ year }} Company Name
  <content></content>
</div>
```


Your final output will be

```html

<div>
  Copyright {{ year }} Company Name
    All rights reserved.
</div>
```

Now with `posthtml-expression` you can pass the value of `year` as local
by keeping the `posthtml-expression` plugin next to this one !

config
```js
posthtml([require('posthtml-modules')({}),expressions({ locals: { year: '2020' } })])...
```


output

```html

<div>
  Copyright 2020 Company Name
    All rights reserved.
</div>
```

**Comment by cossssmin** [comment]
But that's using _global_ variables... The use-case here is for local, _inline_ variables that you define right in the HTML, not in your plugin configuration. Depending on the situation (i.e. public interface), the user might not even have access to define the variable(s) in the plugin's config.

It doesn't make much sense to force the user to add variables that pollute the global scope with JS configuration either, when all that is needed is to pass a variable one time to one module.

Basically, I'm asking for the exact same approach `posthtml-include` uses, with a `locals=""` attribute.

**Comment by anikethsaha** [comment]
ohh.... yeah this makes sense !

posthtml-expression doesn't have run-time flexibility which can be done here using element attributes!

Ok. This can be a good enhancement 👍

**Comment by cossssmin** [comment]
Forked it to start working on this feature, but looks like a lot of things need refactoring:

- `xo` linting is failing - both in the [build](https://travis-ci.org/github/posthtml/posthtml-modules/jobs/231655676) and locally, where I even get errors like:
  - Expected linebreaks to be LF but found CRLF.
  - Unexpected function expression.
  - Comments should not begin with a lowercase character
- testing with `ava` fails with this error:

    ```sh
    × No tests found in test\index.spec.js

    1 uncaught exception

    Uncaught exception in test\index.spec.js


    F:\DEV\open-source\contributing\posthtml-modules\test\index.spec.js:1
    import test from 'ava';
    ^^^^^^

    SyntaxError: Cannot use import statement outside a module
    ```

I can try to refactor the plugin to fix those issues and rewrite the tests so they use fixtures like the other PostHTML plugins - what do you think?

**Comment by anikethsaha** [comment]
yes those need to be fixed and also a dependency upgrade. Can you do that in separate PRs ?

**Comment by cossssmin** [comment]
Sure thing, I'll do the refactor first and then the dependencies, as I need to be able to run tests first.

**Comment by anikethsaha** [comment]
IMO, upgrading dep should be done first cause after updating the tests, and then upgrading dep which might contain breaking changes as this project was last update a while back, may be lead to update the tests again !

**Comment by cossssmin** [comment]
OK, makes sense 👍

**Comment by Scrum** [comment]
@cossssmin @anikethsaha status ?

**Comment by cossssmin** [comment]
Working on it, will try to PR soon
